### DISCUSSION TITLE
Syntax highlighting for shell scripts?

### DISCUSSION THREAD

**Discussion by DVader666** [apache/netbeans#4569] [Q&A]
I don't know if this is an old question. I'm sure I have searched the web for a solution that works for Apache NetBeans 12/13/14, but I only found old modules for old NB 8.2, etc. (they no longer work).

How do I add syntax highlighting for shell scripts like SH, BASH, ZSH? I know there is a plugin (nb-noext-mime-resolver) for returning a proper mime-type from the shebang line and from a file with no extension, but that's just it, no syntax highlighting or anything else (the author is very clear about it).

Why such basic functionality (syntax highlighting for, at least, *nix shell scripts) was never added out of the box?

**Comment by mbien** [comment]
basic syntax highlighting for shell scrpts is part of the old C/C++ cluster and this cluster needs to be updated first before it can be released with NetBeans.

https://github.com/apache/netbeans/tree/cnd

**Comment by larsdesigns** [comment]
Here is a working plugin: https://github.com/larsdesigns/nbshell
